### PR TITLE
Feature: change open api gen process

### LINKED ISSUE #1392: Use pre-generated OpenAPI files
**Is your feature request related to a problem? Please describe.**
It takes time + memory to generate an OpenAPI document. Work has already been done in the following repositories:
- [Microsoft Graph SDK Code Generator](https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator): generate files that follow the conversion settings available
- [Graph metadata](https://github.com/microsoftgraph/msgraph-metadata): store the files that have been generated

**Describe the solution you'd like**
Instead of converting CSDL to OpenAPI, make a call to the already stored files so that we save the time to convert. 
Reuse the other processes involved like caching so that the only thing happening is a call to get the needed openapi file

### PR DESCRIPTION
**Pull request by thewahome**
## Overview

Closes #1392 
Uses the yaml files in the metadata API to return pre-generated open api documents that have already been converted using the conversion settings

### CODE REVIEW

**Review by baywet — DISMISSED** [review]
My only concern is if somebody else than us has taken a dependency on the API /enum values as changing those will break them. Trusting this will at least be coordinated with GE if there are direct dependencies. Otherwise we might want to consider a compatibility mapping.

CC @estruyf if you still have a dependency on devx api for your extensions

**Comment by estruyf** [comment]
@baywet indeed, it is used in the following extension: https://marketplace.visualstudio.com/items?itemName=eliostruyf.vscode-msgraph-autocomplete - one of your colleagues is working on this extension to allow snippets to work as well.

**Comment by thewahome** [comment]
Aah makes sense @baywet ... I will just make a dictionary to map the enum values to the file names ..

**Comment by baywet** [comment]
Thanks for reverting that fast @estruyf It seems the extension is relying on a static zip file instead of calling out to the devx API, did I get this right? https://github.com/estruyf/vscode-msgraph-autocomplete/blob/main/apis.json.gz

**Comment by estruyf** [comment]
@baywet yes, that is the cached version of the API, so that it doesn't always have to call the API, but if it doesn't find it, or there is a new endpoint, it will still call the API.

https://github.com/estruyf/vscode-msgraph-autocomplete/blob/ced76a29162db1f1089f009db3cec95f98c4f8ae/src/utils/AutoComplete.ts#L4

**Comment by baywet** [comment]
Ah great, you're not making use of the style parameter and anyway we're going to be reverting the changes to the API surface. Thanks again for all that input!

File: OpenAPIService/OpenApiService.cs [hunk 1]
```diff
@@ -55,14 +55,20 @@ public class OpenApiService : IOpenApiService
         private readonly int _defaultForceRefreshTime; // time span for allowable forceRefresh of the OpenAPI document
         private readonly Queue<string> _graphUriQueue = new();
         private static readonly RecyclableMemoryStreamManager _streamManager = new();
+        Dictionary<OpenApiStyle, string> fileNames = new Dictionary<OpenApiStyle, string>();
```
Comment (baywet) [inline OpenAPIService/OpenApiService.cs:58]:
```suggestion
        private static readonly Dictionary<OpenApiStyle, string> fileNames = new Dictionary<OpenApiStyle, string>() {
            { OpenApiStyle.GEAutocomplete, "graphexplorer"},
            { OpenApiStyle.Plain, "openapi"},
            { OpenApiStyle.PowerShell, "powershell"},
            { OpenApiStyle.PowerPlatform, "openapi"},
         };
```

File: OpenAPIService/OpenApiService.cs [hunk 2]
```diff
@@ -55,14 +55,20 @@ public class OpenApiService : IOpenApiService
...
                ?? throw new ArgumentNullException(nameof(configuration), $"Value cannot be null: {nameof(configuration)}");
             _defaultForceRefreshTime = FileServiceHelper.GetFileCacheRefreshTime(_configuration[CacheRefreshTimeConfig]);
             _telemetryClient = telemetryClient;
-            _openApiTraceProperties.TryAdd(UtilityConstants.TelemetryPropertyKey_OpenApi, nameof(OpenApiService));            
+            _openApiTraceProperties.TryAdd(UtilityConstants.TelemetryPropertyKey_OpenApi, nameof(OpenApiService));
+
+            fileNames.Add(OpenApiStyle.GEAutocomplete, "graphexplorer");
+            fileNames.Add(OpenApiStyle.Plain, "openapi");
+            fileNames.Add(OpenApiStyle.PowerShell, "powershell");
+            fileNames.Add(OpenApiStyle.PowerPlatform, "openapi");
```
Comment (baywet) [inline OpenAPIService/OpenApiService.cs:71]:
```suggestion
```

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — DISMISSED** [review]

**Review by andrueastman — APPROVED** [review]

**Pull request merged**